proteus/mprans/SpatialTools.py

```diff
@@ -2308,12 +1871,11 @@ def assembleAuxiliaryVariables(domain):
         # ----------------------------
         # RIGID BODIES
         if 'RigidBody' in shape.auxiliaryVariables.keys():
-            body = RigidBody(shape)
+            body = shape.auxiliaryVariables['RigidBody']
             aux['twp'] += [body]
             # fixing mesh on rigid body
             for boundcond in shape.BC_list:
-                boundcond.setMoveMesh(body.last_position, body.h,
-                                      body.rotation_matrix)
+                boundcond.setRigidBodyMoveMesh(body)
             # update the indice for force/moment calculations
             body.i_start = start_flag+1
             body.i_end = start_flag+1+len(shape.BC_list)
```

Comment:
This sets boundary conditions for the rigid body attached to the shape (mesh displacement), and needs to be in SpatialTools when "assembling" the Domain.